### PR TITLE
woo: update

### PR DESCRIPTION
**Pull request by sc0Vu**
```BASH
$ n woo fetchStatus --test
2024-03-01T09:46:13.843Z
Node.js: v18.14.0
CCXT v4.2.56
woo.fetchStatus ()
2024-03-01T09:46:14.540Z iteration 0 passed in 179 ms

{
  status: 'ok',
  info: {
    success: true,
    data: { status: '0', msg: 'System is functioning properly.' },
    timestamp: '1709286374514'
  }
}

$ n woo fetchTime --test
2024-03-01T09:46:43.876Z
Node.js: v18.14.0
CCXT v4.2.56
woo.fetchTime ()
2024-03-01T09:46:44.544Z iteration 0 passed in 178 ms

1709286404517

$ p woo fetchTransfers --test
Python v3.11.3
CCXT v4.2.56
woo.fetchTransfers()
[{'amount': 100000.0,
  'currency': 'USDT',
  'datetime': '2022-08-01T08:19:20.044Z',
  'fromAccount': '0f1bd3cd-dba2-4563-b8bb-0adb1bfb83a3',
  'id': '286',
  'info': {'amount': '100000.00000000',
           'created_time': '1659341960.044',
           'from_application_id': '0f1bd3cd-dba2-4563-b8bb-0adb1bfb83a3',
           'from_user': 'LeverageLow',
           'id': '286',
           'status': 'COMPLETED',
           'to_application_id': 'bdafb16b-51ee-49bc-bd89-cbf43e4e687d',
           'to_user': 'Main',
           'token': 'USDT',
           'updated_time': '1659341960.256'},
  'status': 'ok',
  'timestamp': 1659341960044,
  'toAccount': 'bdafb16b-51ee-49bc-bd89-cbf43e4e687d'},

$ p woo transfer USDT 1000 "0f1bd3cd-dba2-4563-b8bb-0adb1bfb83a3" "c01e6940-a735-4022-9b6c-9d3971cdfdfa" --test
Python v3.11.3
CCXT v4.2.56
woo.transfer(USDT,1000,0f1bd3cd-dba2-4563-b8bb-0adb1bfb83a3,c01e6940-a735-4022-9b6c-9d3971cdfdfa)
{'amount': 1000,
 'currency': 'USDT',
 'datetime': None,
 'fromAccount': '0f1bd3cd-dba2-4563-b8bb-0adb1bfb83a3',
 'id': '47150',
 'info': {'id': '47150', 'success': True},
 'status': 'ok',
 'timestamp': None,
 'toAccount': 'c01e6940-a735-4022-9b6c-9d3971cdfdfa'}
```

### CODE REVIEW

File: ts/src/woo.ts [hunk 1]
```diff
@@ -2152,42 +2213,72 @@ export default class woo extends Exchange {
...
+        //             }
+        //         ],
+        //         "meta": {
+        //             "total": 50,
+        //             "records_per_page": 25,
+        //             "current_page": 1
+        //         },
+        //         "success": true
+        //     }
+        //
```
Comment (carlosmiei) [inline ts/src/woo.ts:2260]: @sc0Vu let's not use `safeValue` please

File: ts/src/woo.ts [hunk 2]
```diff
@@ -2221,8 +2299,8 @@ export default class woo extends Exchange {
             'datetime': this.iso8601 (timestamp),
             'currency': code,
             'amount': this.safeNumber (transfer, 'amount'),
-            'fromAccount': fromAccount,
-            'toAccount': toAccount,
+            'fromAccount': this.safeString (transfer, 'from_application_id'),
```
Comment (carlosmiei) [inline ts/src/woo.ts:2302]: should we use the raw values in the response?

File: ts/src/woo.ts [hunk 3]
```diff
@@ -2152,42 +2213,72 @@ export default class woo extends Exchange {
...
+        }
+        if (since !== undefined) {
+            request['start_t'] = since;
+        }
+        const until = this.safeInteger2 (params, 'until', 'till'); // unified in milliseconds
+        params = this.omit (params, [ 'until', 'till' ]);
+        if (until !== undefined) {
+            request['end_t'] = until;
+        }
+        const response = await this.v1PrivateGetAssetMainSubTransferHistory (this.extend (request, params));
```
Comment (carlosmiei) [inline ts/src/woo.ts:2236]: shouldn't we also support the other way around? 
image

File: ts/src/woo.ts (hunk 2 above)
Comment (sc0Vu) [reply ts/src/woo.ts:2302]: do you mean from account and to account?

File: ts/src/woo.ts (hunk 3 above)
Comment (sc0Vu) [reply ts/src/woo.ts:2236]: Is this api (https://docs.woo.org/#get-asset-history) supposed to use for `fetchTransaction`?

Comment (carlosmiei) [reply ts/src/woo.ts:2236]: yes and can be used as well for fetchDeposits/withdrawals

File: ts/src/woo.ts (hunk 2 above)
Comment (carlosmiei) [reply ts/src/woo.ts:2302]: usually we unify that to "main/spot/etc"

**Pull request merged**